src/crimson/os/seastore/transaction_manager.h

```diff
@@ -301,6 +301,20 @@ class TransactionManager : public ExtentCallbackInterface {
     });
   }
 
+  template <typename T>
+  read_extent_ret<T> get_mutable_extent_by_laddr(Transaction &t, laddr_t laddr, extent_len_t len) {
+    return read_extent<T>(t, laddr, len
+    ).si_then([this, &t](auto&& extent) {
+      auto ext = get_mutable_extent(t, extent)->template cast<T>();
```

Comment:
See the above comment around `create_overwrite()` decision with clone.

For `TM::get_mutable_extent()`, I think it is necessary to generally make sure (by asserting) the extent to be mutated should not be cloned (having refcount > 1) at TM level.

This is another reason that tracking ref-counter in the logical cached extent at runtime might be a better option.

Reply:
If go with the option 1 above, refcount won't be available from the `extent`, so `TM::get_mutable_extent()` cannot validate the refcount.

But `read_extent()` involves a call to `get_pin()`, so the refcount can be validated there after the option 1 is implemented.